### PR TITLE
Inline generation of XRay trace IDs to remove heavyweight SDK depende…

### PR DESCRIPTION
**Pull request by anuraaga**
…ncy.

The X-Ray SDK is pretty heavyweight, meant for instrumenting, not something as simple as generating IDs. This implements ID generation without the dependency. Also removes Guava since it was only used once so not really worth it.

In a future PR I'll reimplement `EC2Resource` as well to remove the AWS SDK dependency too.

### CODE REVIEW

File: sdk_contrib/aws_v1_support/build.gradle [hunk 1]
```diff
@@ -12,12 +12,9 @@ sourceCompatibility = 1.8
 targetCompatibility = 1.8
 
 dependencies {
-    api project(':opentelemetry-api'),
-            project(':opentelemetry-sdk')
```
Comment (bogdandrutu) [inline sdk_contrib/aws_v1_support/build.gradle:16]: This is unnecessary change

File: sdk_contrib/aws_v1_support/build.gradle [hunk 2]
```diff
@@ -12,12 +12,9 @@ sourceCompatibility = 1.8
...
+    api project(':opentelemetry-api')
+    api project(':opentelemetry-sdk')
 
-    implementation libraries.guava
-
-    implementation 'com.amazonaws:aws-xray-recorder-sdk-core:2.4.0'
     implementation 'com.amazonaws:aws-java-sdk-core:1.11.701'
```
Comment (bogdandrutu) [inline sdk_contrib/aws_v1_support/build.gradle:18]: Can be done in on implementation statement

Comment (anuraaga) [reply sdk_contrib/aws_v1_support/build.gradle]: Sure - I took the opposite approach of making them consistent but went with one statement. But I'd argue it's better style to use multiple statements, there's a significant readability loss when the dependencies for the same configuration don't line up vertically IMO. Food for thought.
